src/benzina/torch/operations.py

```diff
@@ -377,11 +375,27 @@ def __call__(self, index, in_shape, out_shape, rng):
         """Return a random similarity transformation."""
         s = np.exp(rng.uniform(low=np.log(self.s[0]), high=np.log(self.s[1])))
         if self.ar is not None:
-            ar = np.exp(rng.uniform(low=np.log(self.ar[0]),
-                                    high=np.log(self.ar[1])))
-            crop_area = s * in_shape[0] * in_shape[1]
-            crop_w = np.sqrt(crop_area * ar)
-            crop_h = np.sqrt(crop_area / ar)
+            for _ in range(10):
+                ar = np.exp(rng.uniform(low=np.log(self.ar[0]),
+                                        high=np.log(self.ar[1])))
+                crop_area = s * in_shape[0] * in_shape[1]
+                crop_w = np.sqrt(crop_area * ar)
+                crop_h = np.sqrt(crop_area / ar)
+
+                if 0 < crop_w <= in_shape[0] and 0 < crop_h <= in_shape[1]:
+                    break
+            else:
+                # Fallback to central crop
+                in_ar = float(in_shape[0]) / float(in_shape[1])
+                if in_ar < self.ar[0]:
+                    crop_w = in_shape[0]
+                    crop_h = int(round(crop_w / self.ar[0]))
+                elif in_ar > self.ar[1]:
+                    crop_h = in_shape[1]
+                    crop_w = int(round(crop_h * self.ar[1]))
+                else:  # whole image
+                    crop_w, crop_h = in_shape
+                self.random_crop = False
```

Comment:
The crop could still be random on one dimension instead but again, this is how torchvision does it in [RandomResizedCrop](https://pytorch.org/docs/stable/_modules/torchvision/transforms/transforms.html#RandomResizedCrop)